### PR TITLE
per-tier reserved beneficiary

### PR DESCRIPTION
**Pull request by mejango**
- tiers are passed in to the constructor as `JB721TierParams` structs.
- tiers are stored as `JBStored721Tier` structs.
- tiers are viewed as `JB721Tier` structs.

### CODE REVIEW

File: contracts/JBTiered721Delegate.sol [hunk 1]
```diff
@@ -100,7 +100,7 @@ contract JBTiered721Delegate is IJBTiered721Delegate, JB721Delegate, Votes, Owna
     return
       JBIpfsDecoder.decode(
         store.baseUriOf(address(this)),
-        store.tierOfTokenId(address(this), _tokenId).data.tokenUri
+        store.tierOfTokenId(address(this), _tokenId).encodedIPFSUri
```
Comment (drgorillamd) [inline contracts/JBTiered721Delegate.sol:103]: Why not making _encodedIpfsUriOf externally accessible and just get store.encodedIpfsUriOf(..) (so we reduce the tierOfTokenId(..) return data size to what's needed)?

File: contracts/JBTiered721DelegateStore.sol [hunk 2]
```diff
@@ -483,7 +587,10 @@ contract JBTiered721DelegateStore is IJBTiered721DelegateStore {
 
         while (_currentSortIndex != 0) {
```
Comment (drgorillamd) [inline contracts/JBTiered721DelegateStore.sol:588]: Shouldn't we exclude an identical active tier already existing (eg
```
if(_tierToAdd.contributionFloor ==
            _storedTierOf[msg.sender][_currentSortIndex].contributionFloor) revert DUPLICATE_FLOOR();
```)

Comment (mejango) [reply contracts/JBTiered721DelegateStore.sol:588]: no problem if multiple tiers have the same floor

File: contracts/JBTiered721Delegate.sol (hunk 1 above)
Comment (mejango) [reply contracts/JBTiered721Delegate.sol:103]: good point!

**Pull request merged**